#!/usr/bin/env python
PACKAGE = "depth_map_fusion"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("line_detection",           bool_t, 0, "Enables line-detection",  False)
gen.add("debug_level",              int_t,  0, "Publish debug-images at level highter than this",  2, 0, 2)
gen.add("threshold",                int_t,  0, "Stereo matching threshold", 20,  0, 255)
gen.add("too_close",                int_t,  0, "Reject pixels closer than too_close", 230,  0, 255)
gen.add("final_blur",               int_t,  0, "How much the fused depth is blurred", 5,  1, 15)
gen.add("score_multiplier",         int_t,  0, "Multiply score to get a brighter image", 56,  0, 255)
gen.add("line_detection_weight",    int_t,  0, "Weight of line-detection vs second-best",  2,  0, 5)
gen.add("sub_mat_max",              int_t,  0, "Maximum score when subtracting second-best",  150,  0, 255)
gen.add("x_offset",                 int_t,  0, "x-axis offset between horizontal and vertical depth-map",   5,  -20, 20)
gen.add("y_offset",                 int_t,  0, "y-axis offset between horizontal and vertical depth-map",  13,  -20, 20)

exit(gen.generate(PACKAGE, "depth_map_fusion", "DepthMapFusion"))